app/services/versioning/file_attachment_revision_updater.rb

```diff
@@ -16,6 +16,10 @@ def assign(fields)
 
       sub_columns = sub_updaters.values.flat_map(&:column_names)
       dup_revision.assign_attributes(fields.except(*sub_columns))
+
+      fields.slice(*sub_updaters.keys).each do |key, value|
```

Comment:
I'm hoping there's a less intrusive way of achieving this. But to recap our discussion, I see the need for the changes reported by the sub-revision updater and the revision updater to be consistent. We should be able to remove this constraint, though, because it's not necessary to have a sub-revision updater for the blob revision, since it can only ever be replaced (as opposed to updated).

To make that a bit clearer, I'm suggesting we:

   - Remove https://github.com/alphagov/content-publisher/blob/0a99bef926d2ed1d9f3a367e9a397fa636ff39d3/app/services/versioning/file_attachment_revision_updater.rb#L38
   - Add `%[blob_revision]` to https://github.com/alphagov/content-publisher/blob/0a99bef926d2ed1d9f3a367e9a397fa636ff39d3/app/services/versioning/file_attachment_revision_updater.rb#L6

Reply:
Sorted.